src/main/java/com/oltpbenchmark/benchmarks/templated/TemplatedBenchmark.java

```diff
@@ -0,0 +1,251 @@
+import com.oltpbenchmark.api.Worker;
+import com.oltpbenchmark.api.templates.ValuesType;
+import com.oltpbenchmark.api.templates.TemplateType;
+import com.oltpbenchmark.api.templates.TemplatesType;
+import com.oltpbenchmark.benchmarks.templated.procedures.GenericQuery;
+import com.oltpbenchmark.benchmarks.templated.procedures.GenericQuery.QueryTemplateInfo;
+import com.oltpbenchmark.benchmarks.templated.util.GenericQueryOperation;
+import com.oltpbenchmark.benchmarks.templated.util.TraceTransactionGenerator;
+import com.opencsv.CSVParser;
+import com.opencsv.CSVParserBuilder;
+
+import jakarta.xml.bind.JAXBContext;
+import jakarta.xml.bind.JAXBElement;
+import jakarta.xml.bind.Unmarshaller;
+
+/**
+ * This class is used to execute templated benchmarks, i.e., benchmarks that
+ * have parameters that the user wants to set dynamically. More information
+ * about the structure of the expected template can be found in the local
+ * readme file.
+ */
+public class TemplatedBenchmark extends BenchmarkModule {
+    private static final Logger LOG = LoggerFactory.getLogger(TemplatedBenchmark.class);
+
+    public TemplatedBenchmark(WorkloadConfiguration workConf) {
+        super(workConf);
+    }
+
+    @Override
+    protected void initClassLoader() {
+        super.initClassLoader();
+
+        this.classLoader = ClassLoader.getSystemClassLoader();
+        if (workConf.getXmlConfig().containsKey("query_templates_file")) {
+            this.classLoader = this.loadQueryTemplates(
+                    workConf.getXmlConfig().getString("query_templates_file"));
+        }
+    }
+
+    @Override
+    protected Package getProcedurePackageImpl() {
+        return (GenericQuery.class.getPackage());
+    }
+
+    @Override
+    protected List<Worker<? extends BenchmarkModule>> makeWorkersImpl() {
+        List<Worker<? extends BenchmarkModule>> workers = new ArrayList<>();
+
+        try {
+            final Map<Class<? extends Procedure>, TraceTransactionGenerator> generators = new HashMap<>();
+            // Create potential parameter bindings for each template. Add those
+            // to a trace transaction generator that will determine how the
+            // parameters are used.
+            for (Entry<TransactionType, Procedure> kv : getProcedures().entrySet()) {
+                // Sanity check that the procedure has the right type.
+                if (!(kv.getValue() instanceof GenericQuery)) {
+                    continue;
+                }
+                GenericQuery proc = (GenericQuery) kv.getValue();
+                QueryTemplateInfo info = proc.getQueryTemplateInfo();
+
+                // Parse parameter values and add each combination to a generator.
+                List<GenericQueryOperation> list = new ArrayList<>();
+                String[] paramsTypes = info.getParamsTypes();
+                for (String binding : info.getParamsValues()) {
+                    CSVParser parser = new CSVParserBuilder()
```

Comment:
Ah, I think I might see the problem.  In the initial loader you're still converting the `<values>` to a comma separated list, and then parsing it back out later, but that has a `, ` instead of just `,` as the separator currently.

I have a fix for that in my branch off of yours plus a few others.

Working on making unit tests work as well so that debugging these are hopefully a little bit easier.

Reply:
Another PR for this against your branch here:
https://github.com/anjagruenheid/benchbase/pull/2

There's a few niggling issues with it (lack of NULL tests), but it appears to be passing for me now.

Reply:
I like it, should I merge it or do you want to do more edits?

Reply:
If you agree, I think you can merge it as is.  I added a few TODOs in it to expand some test coverage (mostly NULL handling), but I think you could handle that too, right?

Reply:
Actually, I fixed a few more things in it and I think it should be fine as is.
Let's take up the lack of support for NULLable params in a future PR and merge this one as is.
Sound good?
I can create a new issue for tracking that.

Reply:
Yep, sounds good!